core/runtime/src/main/java/io/quarkus/artemis/core/runtime/ArtemisDevServicesBuildTimeConfig.java

```diff
@@ -85,26 +87,37 @@ public int getPort() {
     }
 
     public String getImageName() {
-        return imageName;
+        return imageName.orElse("quay.io/artemiscloud/activemq-artemis-broker:1.0.5");
     }
 
     public boolean isShared() {
-        return shared;
+        return shared.orElse(true);
     }
 
     public String getServiceName() {
-        return serviceName;
+        return serviceName.orElse("artemis");
     }
 
     public String getUser() {
-        return user;
+        return user.orElse("admin");
     }
 
     public String getPassword() {
-        return password;
+        return password.orElse("admin");
     }
 
     public String getExtraArgs() {
-        return extraArgs;
+        return extraArgs.orElse("--no-autotune --mapped --no-fsync");
+    }
+
+    public boolean isEmpty() {
```

Comment:
I don't know if I like all those `isEmpty` checks aggregated, very easy to forget one. Do we really need to be optional here? The issue only mentions `url`, I guess it also holds for `xaEnabled`, but I don't see why we need to add this to DevServices as well.

Reply:
The `isEmpty()` - as-is - is the only way I was able to detect reliably what configuration(s) are present at compile-time. For this, we also need to access the runtime properties to detect whether anything is present. This is necessary since we have to call the recorder once per configuration.

Reply:
I see, let's leave it like this for now.